odoo/tools/misc.py

```diff
@@ -1254,13 +1254,55 @@ def __gt__(self, other): return self.val < other.val
     def __le__(self, other): return self.val >= other.val
     def __lt__(self, other): return self.val > other.val
 
+# there is contextlib.suppress since py3
 @contextmanager
 def ignore(*exc):
     try:
         yield
     except exc:
         pass
 
+class reraise_x_as(ContextDecorator):
```

Comment:
Naming seems less than ideal here:

- `as` is duplicated between the name and parameter
- `x` doesn't really make much sense
- `reraise` is generally understood as a raising the original exception (aka a bare `raise`) after having performed some processing

I'd suggest someting like like `replace_exceptions(*_, by=)`, or possibly `substitute` or `translate` instead of `replace`.

An other alternative is `convert`, that is what is used by communities like the Rust community where this sort of thing is routine (e.g. the `try!` macro and `?` operator will convert the error then return it, the end-effect being rather similar to this, the adaptation of the error signal of a callee to that of the current function).

Reply:
Also why not use `contextlib.contextmanager` for this feature?

Reply:
> Also why not use contextlib.contextmanager for this feature?

So that is can be used as a function decorator too

+1 with the naming